trustar/trustar-aws-guard-duty
# En 422/update dependencies

WIP.  Things to do: 
- developer documentation (README). 
- make unittests use a Python testing framework. 
- implement Black / flake8. 
- implement pre-commit hooks. 

---------

What:    Fix GuardDuty integration.  
What it does:   receives guardduty findings from Cloudwatch / EventBridge, submits them as a report to an enclave. 
Configs:  TruSTAR API key, API secret, and enclave ID to submit to. 
How:
- Testing:      
    - create Python 3.7.7 virtualenv.  
    - add your Station creds & enclave ID to the test/private/test.conf file. 
    - activate Python 3.7.7 virtualenv. 
    - add src/ and src/trustar_guardduty_lambda_handler to your IDE's "Source Roots". 
    - run tests/test.py


- Deploying to Lambda:    (in case you want to....)
    - runtime = AmazonLinux2, Python 3.7.7.  
    - SSH into an AmazonLinux2 EC-2 instance. 
    - create Python 3.7.7 virtualenv.   (I used pyenv)
    - activate python 3.7.7 virtualenv.  
    - pip-install trustar --target ~/some_dir
    - clone repo to EC-2 host. 
    - copy contents of src/ dir (recursively) into "some_dir". 
    - zip that dir's contents (.zip file should be ~3.1MB after zipping.) so the contents are at the root of the zip file. 
    - "scp" that zip file back to your local laptop. 
    - use the AWS console browser u/i and the support docs for our Guard Duty integration to create the Lambda. 
    - AWS SAM .yaml file can be used to automate deployment through AWS SAM rather than through the AWS console GUI, if desired  (not tested).   
    -the code in src/lambda_function.py is the execution entrypoint.
